django/urls/base.py

```diff
@@ -179,3 +183,50 @@ def translate_url(url, lang_code):
             else:
                 url = urlunsplit((parsed.scheme, parsed.netloc, url, parsed.query, parsed.fragment))
     return url
+
+
+_PATH_PARAMETER_COMPONENT_RE = re.compile(
+    '<(?:(?P<converter>[^:]+):)?(?P<parameter>[A-Za-z0-9_]+)>'
+)
+
+
+def _route_to_regex(route):
+    parts = ['^']
+    converters = {}
+    while True:
+        match = _PATH_PARAMETER_COMPONENT_RE.search(route)
+        if not match:
+            parts.append(re.escape(route))
+            break
+
+        parts.append(re.escape(route[:match.start()]))
+        route = route[match.end():]
+
+        parameter = match.group('parameter')
+        raw_converter = match.group('converter')
+        if raw_converter is None:
+            # If no converter is specified, the default is ``string``.
+            raw_converter = 'string'
+        converter = get_converter(raw_converter)
+        converters[parameter] = converter
+        parts.append('(?P<' + parameter + '>' + converter.regex + ')')
+    return ''.join(parts), converters
+
+
+def path(route, view, kwargs=None, name=None):
+    regex, converters = _route_to_regex(route)
+    if isinstance(view, (list, tuple)):
+        # For include(...) processing.
+        urlconf_module, app_name, namespace = view
+        return RegexURLResolver(
+            regex,
+            urlconf_module,
+            kwargs,
+            app_name=app_name,
+            namespace=namespace,
+            converters=converters
+        )
+    elif callable(view):
+        return RegexURLPattern(regex + '$', view, kwargs, name, converters=converters)
+    else:
+        raise TypeError('view must be a callable or a list/tuple in the case of include().')
```

Comment:
You can move these additions to a `django.urls.paths` module if you think it's cleaner. I don't have a strong opinion there.